src/configuration-requestor.ts

```diff
@@ -41,22 +47,57 @@ export default class ConfigurationRequestor {
         createdAt: configResponse.createdAt,
       });
 
-      // TODO: different polling intervals for bandit parameters
-      const banditResponse = await this.httpClient.getBanditParameters();
-      if (banditResponse?.bandits) {
-        if (!this.banditModelConfigurationStore) {
-          throw new Error('Bandit parameters fetched but no bandit configuration store provided');
-        }
+      if (this.requiresBanditModelConfigurationStoreUpdate(configResponse.banditReferences)) {
+        const banditResponse = await this.httpClient.getBanditParameters();
+        if (banditResponse?.bandits) {
+          if (!this.banditModelConfigurationStore) {
+            throw new Error('Bandit parameters fetched but no bandit configuration store provided');
+          }
 
-        await this.hydrateConfigurationStore(this.banditModelConfigurationStore, {
-          entries: banditResponse.bandits,
-          environment: configResponse.environment,
-          createdAt: configResponse.createdAt,
-        });
+          await this.hydrateConfigurationStore(this.banditModelConfigurationStore, {
+            entries: banditResponse.bandits,
+            environment: configResponse.environment,
+            createdAt: configResponse.createdAt,
+          });
+        }
       }
     }
   }
 
+  private getLoadedBanditModelVersions(
+    banditModelConfigurationStore: IConfigurationStore<BanditParameters> | null,
+  ) {
+    if (banditModelConfigurationStore === null) {
+      return [];
+    }
+    return Object.values(banditModelConfigurationStore.entries()).map(
+      (banditParam: BanditParameters) => banditParam.modelVersion,
+    );
+  }
+
+  private requiresBanditModelConfigurationStoreUpdate(
+    banditReferences: Record<string, BanditReference>,
+  ): boolean {
+    if (!this.banditModelConfigurationStore) {
+      throw new Error('Bandit parameters fetched but no bandit configuration store provided');
+    }
+    const referencedModelVersions = Object.values(banditReferences).map(
+      (banditReference: BanditReference) => banditReference.modelVersion,
+    );
+
+    const banditModelVersionsInStore = this.getLoadedBanditModelVersions(
```

Comment:
Could this be set when bandits are loaded "set"? I think it's worth it to frontload & store this in memory instead of on each request (typical polling interval is 30s)